### ISSUE TITLE
Unity.Container uses Unity.Abstractions 2.1.1.0 but latest is 2.1.0.0

### ISSUE THREAD

**Issue by patrick-mackay** [unitycontainer/unity#147]
Hi,

I'm getting this error when trying to compile with the latest version.

Error	CS1705	Assembly 'Unity.Container' with identity 'Unity.Container, Version=5.1.1.0, Culture=neutral, PublicKeyToken=489b6accfaf20ef0' uses 'Unity.Abstractions, Version=2.1.1.0, Culture=neutral, PublicKeyToken=6d32ff45e0ccc69f' which has a higher version than referenced assembly 'Unity.Abstractions' with identity 'Unity.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=6d32ff45e0ccc69f'

I cleared the packages folder and the C:\Users\<user>\.nuget\packages folder to get a fresh version of the packages, but still having the problem.

The configuration was Unity 5.1.2, Abstractions 2.1.0 and Interception 5.0.3.

Returning to Unity 5.1.0 and Unity.Abstractions 2.1.0 removes the error.

The line that I think was triggering the issue was, but I'm not sure anymore. Could be that was only a temporary problem since the namespace changes.

var unitySection = (UnityConfigurationSection)configuration.GetSection("unity");
using (var container = new UnityContainer().LoadConfiguration(unitySection)) { <-- THIS

**Comment by patrick-mackay** [comment]
The line that was causing the issue was:

var unitySection = (UnityConfigurationSection)configuration.GetSection("unity");
using (var container = new UnityContainer().LoadConfiguration(unitySection)) { <-- THIS

**Comment by ENikS** [comment]
You lost me there. Are you using composite Unity package or individual Unity.Abstractions and Unity.Container packages?

[Unity.1.2](https://www.nuget.org/packages/Unity/5.1.2) should be self-contained. It includes all core assemblies in the package.

Individual package [Unity.Container5.1.1](https://www.nuget.org/packages/Unity.Container/5.1.1) depends on [Unity.Abstractions.2.1.0](https://www.nuget.org/packages/Unity.Abstractions/2.1.0)

**Comment by patrick-mackay** [comment]
If I remove all but Unity 5.1.0, 

![image](https://user-images.githubusercontent.com/4219141/32297697-b7263b24-bf2e-11e7-9420-8a7aa7574adb.png)

I get this error:

CS0012	The type 'IUnityContainer' is defined in an assembly that is not referenced. You must add a reference to assembly 'Unity.Abstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=6d32ff45e0ccc69f

![image](https://user-images.githubusercontent.com/4219141/32297646-89125826-bf2e-11e7-9e5f-f436cd46d817.png)

**Comment by ENikS** [comment]
Have you included the reference to the assembly? Does it show up as referenced in Solution Explorer if you expand Dependencies section.

The package includes 6 DLLs, you need to make sure all relevant DLLs are referenced.

**Comment by ENikS** [comment]
Do you have a repo I could pull and take a look?

**Comment by patrick-mackay** [comment]
Interesting, I only have 4 dlls, not 6, and when creating the repo, the 6 dlls were registered and no error.

I found in my code why I was also registering Abstrations and Interception alone. Fixed that, removed Abstractions and Interceptions nuget references, updated Unity to 5.1.2, force a reinstallation of it with "Update-Package -reinstall Unity", and after a bunch of updates, now it's working as expected. 

Thanks @ENikS
